### ISSUE TITLE
[Feature] Auto-tagging

### ISSUE THREAD

**Issue by JeffHowden** [DestinyItemManager/DIM#2294]
With set rolls on guns now, we have the opportunity to take preset actions upon certain weapons or armor we get the first time they show up in our inventory (via comparing last known vs now exists). For example, every time I get a Disrespectful Stare or Three Graves, I know I'll want to dismantle them.

It'd be nice if I could define this via an auto-tag tab where I can set up my auto-tag settings. It'd have two major sections. One for weapons and one for armor. Since they'd work identically, I'll describe the weapons section only.

The weapons section would list all the weapons in the game, separated into kinetic, energy, and power. They'd be sorted alphabetically. The UI would provide a multi-selection mechanism. Once the appropriate weapons are selected, I could select the tag I'd like automatically applied to them whenever they show up in my inventory. Optionally, there could be a checkbox to indicate I'd like this auto-tag to run one time against my current inventory.

Coming back to the auto-tag tab, I could switch between tags and the weapons selected would change to indicate previously saved selections, making it possible for me to very simply edit auto-tag settings for a given tag.

Additionally, it may be handy to the user if any any items that aren't in the user's inventory be indicated by the box being dimmed, greyed out, etc.

Obviously, it'd be great if the regular inventory search was available to make finding a particular weapon to set up for an auto-tag just that much easier.

**Comment by JeffHowden** [comment]
This would dovetail nicely with #2288.

**Comment by bhollis** [comment]
This is a neat idea!

**Comment by JeffHowden** [comment]
![image](https://user-images.githubusercontent.com/6680532/31113347-be748e26-a7cd-11e7-9d70-6b3531c64ec5.png)

Using Chrome Inspector, I mocked up my idea to show visually what I'm trying to describe. In my mockup, I've already tagged a number of guns as junk (as shown by the junk tag. I've also selected a few more to be auto-tagged as junk (as shown by the untagged guns with a checkbox that's checked -- 1 energy and 2 power).

Whenever my overall inventory changes due to picking up legendary/luminous rank up rewards from vendors, decrypting legendary and exotic engrams at the cryptarch, etc. and DIM notices this change, it checks to see if there are any auto-tags defined. If so, it runs through each auto-tag directive and checks if any of the *new* guns were marked for auto-tagging. If so, it automatically applies the tag I've chosen to be applied. So, if I got a Widow's Bite, it'd automatically be tagged as junk.
